lib/telepoison.ex

```diff
@@ -35,20 +35,67 @@ defmodule Telepoison do
     :otel_propagator_text_map.inject(headers)
   end
 
-  def request(%Request{options: opts} = request) do
-    save_parent_ctx()
-    span_name = Keyword.get_lazy(opts, :ot_span_name, fn -> compute_default_span_name(request) end)
+  @doc """
+  Performs a request using Telepoison with the provided `request` options.
+
+  See `HTTPoison.request/1` for further details regarding `request` options.
+
+  Will attempt to automatically set the `http.route` Open Telemetry metadata attribute by
+  removing the last part of the `request.url`, since this part typically contains dynamic data.
+
+  If this behavior is not desirable, it can be set directly by using the `:resource_route` optional parameter.
+
+    ## Examples
+
+      iex> request = %HTTPoison.Request{
+      ...> method: :post,
+      ...> url: "https://www.example.com/users/edit/2",
+      ...> body: "{\"foo\": 3}",
+      ...> headers: [{"Accept", "application/json"}]}
+      iex> Telepoison.request(request, resource_route: "users/edit/:user_id")
+      {:ok, response}
+
+  """
+  def request(%Request{options: request_opts} = request, opts \\ []) do
+    span_name = Keyword.get_lazy(request_opts, :ot_span_name, fn -> compute_default_span_name(request) end)
+
+    [env: env, service: service] = Logger.metadata() |> Keyword.take([:service, :env])
```

Comment:
I don't think Telepoison should rely on any specific Logger metadata to having been set!

IMO Telepoison shouldn't try to find and set `service` and `env` at all. The callers can already specify any additional attribute they want to have by passing it to calls to Telepoison.

And for these "global tags" that are always fixed in a deployment, the opentelemetry exporter and opentelemetry collector can also be configured to automatically add some global tags like service and env to all traces.

Reply:
Removed!